velocitas_lib/__init__.py

```diff
@@ -181,14 +189,14 @@ def obtain_local_file_path(
         download_path = os.path.join(
             get_project_cache_dir(), "downloads", path_or_uri.split("/")[-1]
         )
-    if os.path.isfile(download_path):
-        path, file = os.path.split(download_path)
-        parts = file.split(".", 1)
-        filename = f"{parts[0]}_1.{parts[1]}" if len(parts) > 1 else f"{parts[0]}_1"
 
-        download_path = os.path.join(path, filename)
+    try:
+        download_file(path_or_uri, download_path)
+    except requests.ConnectionError:
+        if have_internet_connection() or not os.path.exists(download_path):
```

Comment:
if the file can't be downloaded to whatever reason but we have internet connection the exception is also raised if the file is available.

Should we maybe split this check to only through if the file is not available? 
```suggestion
        if have_internet_connection() and not os.path.exists(download_path):
```

Reply:
This is by intention: If there is an online connection, we should always try to download the file and give an error if the download fails - regardless if the file is present